epoxy-adapter/src/main/java/com/airbnb/epoxy/DiffHelper.java

```diff
@@ -189,9 +189,16 @@ private UpdateOpHelper buildDiff(UpdateOpHelper updateOpHelper) {
     collectMoves(updateOpHelper);
     collectChanges(updateOpHelper);
 
+    resetOldState();
+
     return updateOpHelper;
   }
 
+  private void resetOldState() {
+    oldStateList.clear();
```

Comment:
notice that `prepareStateForDiff` calls this too. it seems like we don't need to do it there if we do it here instead.

Reply:
Yes, but it was delayed to next `buildModels` call. After the `buildModels` is called, oldStateList is cleared, but it hold new oldStateList instead. It always hold 2x instances (oldStateList and currentStateList).

Reply:
Just like, in first time, it hold state-v1 (as oldStateList) and state-v2 (as currentStateList), after next buildModels call, it hold state-v2 (as oldStateList) and state-v3 (as currentStateList).